packages/x-data-grid/src/hooks/features/rowSelection/useGridRowSelection.ts

```diff
@@ -453,9 +453,6 @@ export const useGridRowSelection = (
   const isFirstRender = React.useRef(true);
   const removeOutdatedSelection = React.useCallback(
     (sortModelUpdated = false) => {
-      if (isFirstRender.current) {
```

Comment:
`isFirstRender` is not used anymore. it can be cleared from the file

with that
how does this affect https://github.com/mui/mui-x/pull/14909
since that was the only change added to address that issue

Reply:
Thanks, cleaned up `isFirstRender`.

> how does this affect https://github.com/mui/mui-x/pull/14909

#15184 removed the root cause for #14909, `isFirstRender` ref was not needed anymore for that fix.